src/System.Management.Automation/engine/parser/SemanticChecks.cs

```diff
@@ -1168,6 +1168,19 @@ public override AstVisitAction VisitInvokeMemberExpression(InvokeMemberExpressio
             return AstVisitAction.Continue;
         }
 
+        public override AstVisitAction VisitIndexExpression(IndexExpressionAst indexExpressionAst)
+        {
+            if (indexExpressionAst.NullConditional && indexExpressionAst.Parent is AssignmentStatementAst)
```

Comment:
There are assignments like this `($a, $b) = ...`, and this won't cover that case.
Any other forms of assignments? Need to talk to @JamesWTruher